crates/ruff/src/resolve.rs

```diff
@@ -91,7 +106,18 @@ pub fn resolve(
     // "closest" `pyproject.toml` file for every Python file later on, so these act
     // as the "default" settings.)
     debug!("Using Ruff default settings");
-    let config = config_arguments.transform(Configuration::default());
+    let mut config = config_arguments.transform(Configuration::default());
+    if config.target_version.is_none() {
+        let fallback = find_fallback_target_version(
+            stdin_filename
+                .as_ref()
+                .unwrap_or(&path_dedot::CWD.as_path()),
+        );
+        if fallback.is_some() {
+            debug!("Deriving `target-version` from found `requires-python`");
+        }
+        config.target_version = fallback.map(Into::into);
```

Comment:
nit: It's unclear to me what does this convert into, it might be useful to explicitly use the `from` syntax